infra/pipeline.ts

```diff
@@ -0,0 +1,153 @@
+    });
+
+    const platforms = [
+      {'arch': 'amd64', 'buildImage': codebuild.LinuxBuildImage.AMAZON_LINUX_2_3},
+      {'arch': 'arm64', 'buildImage': codebuild.LinuxBuildImage.AMAZON_LINUX_2_ARM},
+    ];
+
+    // Create build and verify project for each platform
+    for (const platform of platforms) {
+      const arch = platform['arch'];
+
+      const buildProject = new codebuild.PipelineProject(this, `BuildImage-${arch}`, {
+        buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec.yml'),
+        environment: {
+          buildImage: platform['buildImage'],
+          privileged: true,
+          environmentVariables: {
+            ARCH_SUFFIX: { value: arch },
+          }
+        }
+      });
+
+      const verifyProject = new codebuild.PipelineProject(this, `VerifyImage-${arch}`, {
+        buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec_verify.yml'),
+        environment: {
+          buildImage: platform['buildImage'],
+          privileged: true,
+          environmentVariables: {
+            ARCH_SUFFIX: { value: arch },
+          }
+        }
+      });
+
+      buildProject.addToRolePolicy(new iam.PolicyStatement({
+        actions: ["ecr:GetAuthorizationToken",
+          "ecr:BatchCheckLayerAvailability",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:GetRepositoryPolicy",
+          "ecr:DescribeRepositories",
+          "ecr:ListImages",
+          "ecr:DescribeImages",
+          "ecr:BatchGetImage",
+          "ecr:InitiateLayerUpload",
+          "ecr:UploadLayerPart",
+          "ecr:CompleteLayerUpload",
+          "ecr:PutImage",
+          "ecr-public:*",
+          "secretsmanager:GetSecretValue",
+          "sts:GetServiceBearerToken",
+          "sts:AssumeRole",
+        ],
+        resources: ["*"]
+      }));
+
+      verifyProject.addToRolePolicy(new iam.PolicyStatement({
+        actions: ["ecr:GetAuthorizationToken",
+          "ecr:BatchCheckLayerAvailability",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:GetRepositoryPolicy",
+          "ecr:DescribeRepositories",
+          "ecr:ListImages",
+          "ecr:DescribeImages",
+          "ecr:BatchGetImage",
+          "ecr:InitiateLayerUpload",
+          "ecr:UploadLayerPart",
+          "ecr:CompleteLayerUpload",
+          "ecr:PutImage",
+          "secretsmanager:GetSecretValue",
+        ],
+        resources: ["*"]
+      }));
+
+      const buildAction = new actions.CodeBuildAction({
+        actionName: `Build-${platform['arch']}`,
+        project: buildProject,
+        input: sourceOutput
+      });
+
+      const verifyAction = new actions.CodeBuildAction({
+        actionName: `Verify-${platform['arch']}`,
+        project: verifyProject,
+        input: sourceOutput
+      });
+
+      // Add build action for each platform to the build stage
+      buildStage.addAction(buildAction);
+      verifyStage.addAction(verifyAction);
+    }
+  }
+
+  // TODO Build stage for creating manifest
```

Comment:
do we care about this? maybe we can just delete this todo

Reply:
It was mentioned as a fast-follow action -- aws-for-fluent-bit includes the manifest creation as part of their [buildspec](https://github.com/aws/aws-for-fluent-bit/blob/mainline/buildspec.yml#L28-L32), and it's mentioned also as a recommended practice in this AWS devops [blog post](https://aws.amazon.com/blogs/devops/creating-multi-architecture-docker-images-to-support-graviton2-using-aws-codebuild-and-aws-codepipeline/).